File: tests/benchmarks/python/strip_brackets.py
#!/usr/bin/env python

import utils, sys

def strip_brackets(string):
    d = 0
    out = ''
    for c in string:
        if c == '{' or c == '[': d += 1

        if d > 0:
            out += ' '
        else:
            out += c

        if c == '}' or c == ']': d -= 1

    return out

for f in sys.argv[1:]:
    t = utils.benchmark(lambda: utils.with_utf8_file(f, strip_brackets))
    sys.stderr.write('{0}: {1}\n'.format(f, t))

File: tests/benchmarks/ruby/strip_brackets.rb
#!/usr/bin/env ruby

require './utils.rb'

def strip_brackets(str)
  d = 0
  out = ''

  str.each_char do |c|
    d += 1 if c == '{' || c == '['
    out << if d > 0 then ' ' else c end
    d -= 1 if c == '}' || c == ']'
  end

  out
end

ARGV.each do |f|
  t = benchmark { with_utf8_file(f) { |c| strip_brackets(c) } }
  STDERR.puts "#{f}: #{t}"
end
